app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx

```diff
@@ -0,0 +1,134 @@
+
+  const getOngoingData = () => {
+    MyDataPrepApi.getWorkspaceList({
+      context: 'default',
+    }).subscribe((res) => {
+      res.values.forEach((item) => {
+        const params = {
+          context: 'default',
+          workspaceId: item.workspaceId,
+        };
+        const requestBody = {
+          directives: item.directives,
+          limit: 1000,
+          insights: {
+            name: item.name,
+            workspaceName: item.workspaceName,
+            path: item?.sampleSpec?.path,
+            visualization: {},
+          },
+        };
+        MyDataPrepApi.execute(params, requestBody).subscribe((response) => {
+          let dataQuality = 0;
+          response.headers.forEach((head) => {
+            const general = response.summary.statistics[head].general;
+            const { empty: empty = 0, 'non-null': nonEmpty = 100 } = general;
+            const nonNull = Math.floor((nonEmpty - empty) * 10) / 10;
+            dataQuality = dataQuality + nonNull;
+          });
+
+          const totalDataQuality = dataQuality / response.headers.length;
+
+          setOngoingExpDatas((current) => [
+            ...current,
+            {
+              connectionName:
+                item?.sampleSpec?.connectionName === undefined
+                  ? 'Upload'
+                  : item?.sampleSpec?.connectionName,
+              workspaceName: item.workspaceName,
+              recipeSteps: item.directives.length,
+              dataQuality: totalDataQuality,
+            },
+          ]);
+        });
+      });
+    });
+  };
+
+  useEffect(() => {
+    getOngoingData();
+  }, []);
+
+  useEffect(() => {
+    const final = updatedData(ongoingExpDatas);
+    setFinalArray(final);
+  }, [ongoingExpDatas]);
+
+  return (
+    <Box data-testid="ongoing-data-explore-parent">
+      {finalArray.map((item) => {
+        return (
+          <Link
+            to={`/ns/${getCurrentNamespace()}/wrangler-grid/:${`${'testDataset'}`}`}
+            style={{ textDecoration: 'none' }}
+          >
+            <Grid container className={classes.gridContainer}>
+              {item.map((eachItem) => {
+                switch (eachItem.type) {
+                  case 'iconWithText':
+                    return (
+                      <Grid item className={classes.elementStyle}>
+                        <Box className={classes.iconStyle}> {eachItem.icon}</Box>
+
+                        <Typography variant="body1">{eachItem.label}</Typography>
+                      </Grid>
+                    );
+                  case 'text':
+                    return (
+                      <Grid item className={classes.elementStyle}>
+                        <Typography variant="body1"> {eachItem.label}</Typography>
+                      </Grid>
+                    );
+                  case 'percentageWithText':
+                    const percent = parseInt(eachItem.label);
+
+                    return (
```

Comment:
Can the `Typography` elements be combined into one? Also, is there a visual difference beyond just the color change? We should not rely on color alone as it's not accessible to color-blind users.

Reply:
discussed in the call and to be retained as these are in separate cells